### ISSUE TITLE
Consider Bumping Version of ws

### ISSUE THREAD

**Issue by erichosick** [ably/ably-js#1464]
TLDR: Recommend bumping `ws` version.

The older version of ws (^5.1) (see [ably-js/package.json](https://github.com/ably/ably-js/blob/83b45167e1bf03ab28b12924362ebcdcd1c3a353/package.json#L32C6-L32C8) is incompatible with newer versions of ws. 

In pnpm workspaces, this was causing issues trying to run ably-js along with packages that use the newer `ws` package: (such as grafserv) in a pnpm workspace.

The error thrown was:

```
TypeError: WebSocketServer is not a constructor
    at makeNodeUpgradeHandler
```

because the newer ws expose WebSocketServer as follows:

```
    const ws = await import("ws");
    const { WebSocketServer } = ws;
```

It is possible to bypass this by using a `.pnpmfile.cjs`:

```
// .pnpmfile.cjs
module.exports = {
  hooks: {
    readPackage(packageJson) {

      // Exception has occurred: TypeError: WebSocketServer is not a constructor
      // because /ably@1.2.45(react-dom@18.2.0)(react@18.2.0):
      // uses ws: 5.2.3 
      if (
        packageJson.dependencies &&
        packageJson.dependencies['ws']
      ) {
        packageJson.dependencies['ws'] = '8.14.2';
      }

      if (
        packageJson.peerDependencies &&
        packageJson.peerDependencies['ws']
      ) {
        packageJson.peerDependencies['ws'] = '8.14.2';
      }

      return packageJson;
    }
  },
}
```

but overall this was hard to track down so bumping the version would make it easier for users of ably-js.

**Comment by erichosick** [comment]
Thank you!

**Comment by owenpearson** [comment]
@erichosick no worries, thanks for reporting this! We've just released version 1.2.46 which includes the newer version of `ws` 🙂
